### PR TITLE
auto-generate cloud-init and curtin content for dev-summary

### PR DESCRIPTION
**Pull request by blackboxsw**

### CODE REVIEW

File: generate.sh [hunk 1]
```diff
@@ -18,6 +18,13 @@ if [ -e "$SUMMARY_FILEPATH" ]; then
     rm "$SUMMARY_FILEPATH"
 fi
 
+
+if [ -z  `which log2dch` ]; then
+  echo "Missing log2dch script from https://github.com/CanonicalLtd/uss-tableflip for cloud-init and curtin dev-summary"
+  exit 1
```
Comment (powersj) [inline generate.sh]: Does this mean I need to have uss-tableflip/scripts in my path at all times?

File: generate.sh [hunk 2]
```diff
@@ -18,6 +18,13 @@ if [ -e "$SUMMARY_FILEPATH" ]; then
...
+fi
+
 cp template.md "$SUMMARY_FILEPATH"
+./report_git_backlog.py $LAST_SUMMARY_DATE $SUMMARY_FILEPATH
```
Comment (powersj) [inline generate.sh]: Can we quote these

**Review by powersj — COMMENTED** [review]
2 comments

File: generate.sh (hunk 2 above)
Comment (blackboxsw) [reply generate.sh]: done

File: generate.sh (hunk 1 above)
Comment (blackboxsw) [reply generate.sh]: yeah, at least it's needed to generate the dynamic content from the git commit messages.  Do you want it to be optional? we can do that too.

File: generate.sh [hunk 3]
```diff
@@ -18,6 +18,12 @@ if [ -e "$SUMMARY_FILEPATH" ]; then
     rm "$SUMMARY_FILEPATH"
 fi
 
+
+if [ -z  `which log2dch` ]; then
+  echo "Warning: missing log2dch script from https://github.com/CanonicalLtd/uss-tableflip for cloud-init and curtin dev-summary. This will result in missing bug references on published commit messages."
```
Comment (smoser) [inline generate.sh:23]: this is fine.... but fwiw, 'command -v' is a shell builtin.  and both command -v and 'which' exit non-zero if the command is not in the path.

so:
  if command -v log2dch >/dev/null; then
     echo "Warning"....
   fi

**Pull request merged**